kv/memberlist/dnsprovider.go

```diff
@@ -0,0 +1,18 @@
+package memberlist
+
+import (
+	"context"
+)
+
+// DNSProvider supports storing or resolving a list of addresses.
+type DNSProvider interface {
+	// Resolve stores a list of provided addresses or their DNS records if requested.
+	// Addresses prefixed with `dns+` or `dnssrv+` will be resolved through respective DNS lookup (A/AAAA or SRV).
```

Comment:
I think use "dns+" prefixes is an implementation detail of Thanos resolver. I wouldn't consider it part of interface contract.

I would say instead that addresses are supplied by user, and implementations may interpret these addresses in their own specific ways.

Reply:
Thanks @pstibrany .